packages/aws-amplify-react-native/src/InAppMessaging/components/FullScreenMessage/FullScreenMessage.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2017-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+import React, { useMemo } from 'react';
+import { Image, StyleSheet, Text, View, ViewStyle } from 'react-native';
+import isEmpty from 'lodash/isEmpty';
+
+import icons from '../../../icons';
+import { useInAppMessageButtonStyle, useInAppMessageImage } from '../../hooks';
+import { Button, IconButton } from '../../ui';
+
+import { ICON_BUTTON_HIT_SLOP, ICON_BUTTON_SIZE } from '../constants';
+import MessageWrapper from '../MessageWrapper';
+import { styles } from './styles';
+import { FullScreenMessageProps } from './types';
+
+export default function FullScreenMessage({
+	body,
+	container,
+	header,
+	image,
+	layout,
+	onClose,
+	primaryButton,
+	secondaryButton,
+	style,
+}: FullScreenMessageProps) {
+	const { imageStyle, shouldDelayMessageRendering, shouldRenderImage } = useInAppMessageImage(image, layout);
+	const { primaryButtonStyle, secondaryButtonStyle } = useInAppMessageButtonStyle({
+		baseButtonStyle: styles,
+		messageButtonStyle: { primaryButton: primaryButton?.style, secondaryButton: secondaryButton?.style },
+		overrideButtonStyle: style,
+	});
+
+	const hasPrimaryButton = !isEmpty(primaryButton);
+	const hasSecondaryButton = !isEmpty(secondaryButton);
+
+	// due to limitations with passing spacing style to the underlying style prop of the SafeAreaView of the
+	// MessageWrapper, only pass backgroundColor as a message or override style, apply all other override style
+	// values to container View
+	const { containerStyle, wrapperStyle } = useMemo(() => {
+		const { container: baseOverrideContainerStyle } = style ?? {};
+		const { backgroundColor, ...overrideContainerStyle } = Array.isArray(baseOverrideContainerStyle)
+			? StyleSheet.flatten(baseOverrideContainerStyle)
+			: (baseOverrideContainerStyle as ViewStyle) ?? {};
```

Comment:
You should be able to just pass style to flatten
```suggestion
		const { backgroundColor, ...overrideContainerStyle } = StyleSheet.flatten(baseOverrideContainerStyle ?? {});
```

Reply:
That's correct, using the ternary to prevent an unnecessary call to `StyleSheet.flatten`

Reply:
Is `style` a StyleSheet? If it's not, I'm not sure it's worth sacrificing the readability and having to add an isArray check to save a call to flatten considering flatten is essentially just going to pass it straight through at that point.

Reply:
Not entirely convinced that the ternary creates a lack of readability. 

Since this is going to run in a client's application my intuition is to avoid calling a method that in its own documentation warns that over use can tax the application memory. Granted this use case is not that, but just trying to be mindful